pilot/pkg/model/push_context.go

```diff
@@ -371,6 +442,10 @@ func (ps *PushContext) Services(proxy *Proxy) []*Service {
 	// Second add public services
 	out = append(out, ps.publicServices...)
 
+	if proxy != nil && proxy.Type != SidecarProxy {
+		return ps.FilteredGatewayServices(proxy, out)
+	}
+
```

Comment:
And multi cluster is also broken.

Reply:
the gateway for multi cluster use will have filtering disabled.